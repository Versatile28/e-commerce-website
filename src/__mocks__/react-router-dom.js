const React = require('react');

module.exports = {
  MemoryRouter: ({ children }) => <>{children}</>,
  Link: ({ to, children, ...rest }) =>
    React.createElement('a', { href: to, 'data-testid': 'mock-link', ...rest }, children),
};
