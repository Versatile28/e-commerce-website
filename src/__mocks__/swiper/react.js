module.exports = {
   Swiper: function Swiper(props) { return props.children; },
   SwiperSlide: function SwiperSlide(props) { 
     return <div data-testid="swiper-slide">{props.children}</div>; 
   }
 };