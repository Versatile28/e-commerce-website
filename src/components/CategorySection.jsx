import React from 'react'
import { Container, Row, Col } from 'react-bootstrap';
import { Link } from 'react-router-dom';

export default function CategorySection() {
  return (
<div className='category-section'>
<Container className="w-xs category-container">
   <Row>
     <Col xs={12} sm={6} className="mb-3 mb-sm-0">
       <div className="image-container">
         <Link to='/category-full'>
           <img
             src="images/category-women.webp"
             alt="Women Category"
             className="img-fluid"
           />
           <div className="overlay-text">Women</div>
         </Link>
       </div>
     </Col>

     <Col xs={12} sm={6}>
       <div className="image-container">
         <Link to='/category-full'>
           <img
             src="images/category-men.webp"
             alt="Men Category"
             className="img-fluid"
           />
           <div className="overlay-text">Men</div>
         </Link>
       </div>
     </Col>
   </Row>
 </Container>
</div>
  )
}
