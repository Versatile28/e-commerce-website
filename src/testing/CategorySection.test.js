import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CategorySection from '../components/CategorySection';

describe('CategorySection Component', () => {
  it('renders two category links to /category-full', () => {
    render(
      <MemoryRouter>
        <CategorySection />
      </MemoryRouter>
    );

    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(2);
    links.forEach((link) => {
      expect(link).toHaveAttribute('href', '/category-full');
    });
  });

  it('renders women category image with correct src and alt text', () => {
    render(
      <MemoryRouter>
        <CategorySection />
      </MemoryRouter>
    );

    const womenImg = screen.getByRole('img', { name: 'Women Category' });
    expect(womenImg).toBeInTheDocument();
    expect(womenImg).toHaveAttribute('src', 'images/category-women.webp');
  });

  it('renders men category image with correct src and alt text', () => {
    render(
      <MemoryRouter>
        <CategorySection />
      </MemoryRouter>
    );

    const menImg = screen.getByRole('img', { name: 'Men Category' });
    expect(menImg).toBeInTheDocument();
    expect(menImg).toHaveAttribute('src', 'images/category-men.webp');
  });

  it('displays overlay text for both categories', () => {
    render(
      <MemoryRouter>
        <CategorySection />
      </MemoryRouter>
    );

    expect(screen.getByText('Women')).toBeVisible();
    expect(screen.getByText('Men')).toBeVisible();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CategorySection />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
