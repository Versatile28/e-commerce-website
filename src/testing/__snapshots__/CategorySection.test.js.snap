// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySection Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="category-section"
  >
    <div
      class="w-xs category-container container"
    >
      <div
        class="row"
      >
        <div
          class="mb-3 mb-sm-0 col-sm-6 col-12"
        >
          <div
            class="image-container"
          >
            <a
              data-testid="mock-link"
              href="/category-full"
            >
              <img
                alt="Women Category"
                class="img-fluid"
                src="images/category-women.webp"
              />
              <div
                class="overlay-text"
              >
                Women
              </div>
            </a>
          </div>
        </div>
        <div
          class="col-sm-6 col-12"
        >
          <div
            class="image-container"
          >
            <a
              data-testid="mock-link"
              href="/category-full"
            >
              <img
                alt="Men Category"
                class="img-fluid"
                src="images/category-men.webp"
              />
              <div
                class="overlay-text"
              >
                Men
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
